test/cljx/schema/core_test.cljx

```diff
@@ -229,6 +239,21 @@
     (is (= '(either {:a Int} Int) (s/explain (s/either {:a s/Int} s/Int))))
     (is (s/explain schema))))
 
+(deftest constrained-test
+  (let [schema (s/constrained
+                {s/Keyword s/Keyword}
+                (fn [m] (every? (fn [[k v]] (= k v)) m))
+                'equal-keys?)]
+    (valid! schema {})
+    (valid! schema {:foo :foo :bar :bar})
+    (invalid! schema {"foo" "foo"})
+    (invalid! schema {:foo :bar} "(not (equal-keys? {:foo :bar}))")
+    (invalid! schema {:foo 1} "(not (equal-keys? {:foo 1}))")
```

Comment:
yeah, but you're right that they could be unified.  I'm not sure if there's an easy way to get predicate names into conditional schema, that would also help with the error messages there though so is worth thinking about.  ideas?